am-core-web-service/src/main/java/org/am/core/web/service/admingeneral/ClassroomService.java

```diff
@@ -0,0 +1,113 @@
+package org.am.core.web.service.admingeneral;
+
+import org.am.core.web.domain.entity.admingeneral.Classroom;
+import org.am.core.web.dto.admingeneral.ClassroomDto;
+import org.am.core.web.dto.admingeneral.ClassroomRequest;
+import org.am.core.web.repository.jpa.CustomMap;
+import org.am.core.web.repository.jpa.admingeneral.ClassroomRepository;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+@Transactional
+public class ClassroomService implements CustomMap<ClassroomDto, Classroom> {
+    private final ClassroomRepository classroomRepository;
+
+    public ClassroomService(ClassroomRepository classroomRepository) {
+        this.classroomRepository = classroomRepository;
+    }
+
```

Comment:
If you are using @Transactional on Class level, all method inherit it and by default is readOnly=false. For all reading table you have to change @Transactional(readOnly = true) . AreaService.java